### PR TITLE
Extract CSS resets to reusable mixins

### PR DESCRIPTION
**Pull request by youknowriad**
Global CSS resets are a bad idea and the more we move away from them, the better is for reusability. 
That said, the current resets we have are not very well organized and can't be reused across pages.

With the new widgets page, we need to share these resets between the editor and the widgets page. This PR extract these resets (not all of them) to two reusable mixins:

 - **reset** A context-agnostic reset
 - **wp-admin-reset** A way to prepare a WordPress Admin Page to be Gutenberg based (or JS UI based)

**Testing instructions**

 - Ensure the different Gutenberg UI elements (especially inputs and textareas...) are still showing as expected.

### CODE REVIEW

File: assets/stylesheets/_mixins.scss [hunk 1]
```diff
@@ -343,3 +343,219 @@
...
+@mixin reset {
+	box-sizing: border-box;
+
+	*,
+	*::before,
+	*::after {
+		box-sizing: inherit;
+	}
+
+	select {
```
Comment (youknowriad) [inline assets/stylesheets/_mixins.scss]: Ideally for reusable components using inputs/selects... we shouldn't rely on the existence of these resets because this "reset" mixins are not included in the styles of these reusable components which means if you use a given component outside of the Gutenberg context, it may not be shown properly.

This is a separate task though that should be done step by step.

File: packages/editor/src/components/post-title/style.scss [hunk 2]
```diff
@@ -24,6 +24,7 @@
 		border: $border-width solid transparent;
 		border-left-width: 0;
 		border-right-width: 0;
+		border-radius: 0;
```
Comment (kjellr) [inline packages/editor/src/components/post-title/style.scss]: In my testing (FF, Chrome, Safari for Mac), this `border-radius` rule seems to be the only change that's needed in this file: if I remove the changes on `L10` & `L53-64`, this still seems to work just fine. Are we sure we need those updates?

File: assets/stylesheets/_mixins.scss [hunk 3]
```diff
@@ -343,3 +343,219 @@
...
+				}
+			}
+
+			&:focus {
+				box-shadow: 0 0 0 2px $dark-gray-500;
+			}
+		}
+	}
+
+	input[type="radio"] {
```
Comment (kjellr) [inline assets/stylesheets/_mixins.scss]: I think we're missing some small-screen rules here: I just noticed our radio controls look a little weird on mobile. 😄 But that existed before this PR, and can be handled separately.

Screen Shot 2019-03-20 at 5 14 26 PM

**Review by kjellr — COMMENTED** [review]
This works well in my testing! Just left a couple super-minor comments inline. Once we double check that post title stuff, this should be good to go. 👍

File: packages/editor/src/components/post-title/style.scss (hunk 2 above)
Comment (youknowriad) [reply packages/editor/src/components/post-title/style.scss]: L10 is probably not needed, I'll check but the focus style is needed in the "Spotlight Mode"

**Comment by youknowriad** [comment]
Can I get a ✅ here :)

**Review by kjellr — APPROVED** [review]
LGTM! Thanks, @youknowriad.

**Comment by youknowriad** [comment]
Thanks for the review @kjellr

File: packages/editor/src/components/post-title/style.scss [hunk 4]
```diff
@@ -49,6 +50,18 @@
 		&:-ms-input-placeholder {
 			color: $dark-opacity-200;
 		}
+
+		&:focus {
+			// Stack borders on mobile.
+			border: $border-width solid transparent;
+			border-left-width: 0;
+			border-right-width: 0;
```
Comment (aduth) [inline packages/editor/src/components/post-title/style.scss]: This appears to be the cause for a visual regression in the focus style of the title.

Note the lack of right border:

image

Comment (youknowriad) [reply packages/editor/src/components/post-title/style.scss]: 👍 PR to fix it https://github.com/WordPress/gutenberg/pull/14572

**Comment by afercia** [comment]
Noting that with this change, all the rules related to inputs now apply also to the metaboxes area and to the custom fields area, see  for example https://github.com/Yoast/wordpress-seo/issues/13316

The reason is simple:
- before: all the input rules were scoped to specific parts of the Gutenberg UI: 
```
.editor-post-permalink,
.edit-post-sidebar,
.editor-post-publish-panel,
.block-editor-block-list__block,
.components-popover,
.components-modal__content 
```
- after: they're in the new mixin which is scoped to `.block-editor` and `.block-editor` is the Gutenberg main container, which includes both the metaboxes and custom fields areas.

In other words, this change breaks UIs provided by plugins. It should either be communicated to plugin authors well in advance so they can adjust their CSS or (preferably) the new mixin should be better scoped.
